Modules/_io/fileio.c

```diff
@@ -447,6 +447,30 @@ _io_FileIO___init___impl(fileio *self, PyObject *nameobj, const char *mode,
         }
     }
     else {
+#if defined(S_ISREG) && defined(ENOTDIR)
+        /* On AIX and Windows, open may succeed for files with a trailing slash.
+           The Open Group specifies filenames ending with a trailing slash should
+           be an error - ENOTDIR */
+	if (S_ISREG(fdfstat.st_mode)) {
```

Comment:
Change this condition to `fd < 0 && S_ISREG(fdfstat.st_mode)` to ignore opening a file descriptor, so we know `widename` / `name` aren't `NULL`.

Reply:
a) If fd < 0 I do not think we are even in this block. See lines 432-449
  +432      Py_BEGIN_ALLOW_THREADS
  +433      fstat_result = _Py_fstat_noraise(self->fd, &fdfstat);
  +434      Py_END_ALLOW_THREADS
  +435      if (fstat_result < 0) {
  +436          /* Tolerate fstat() errors other than EBADF.  See Issue #25717, where
  +437          an anonymous file on a Virtual Box shared folder filesystem would
  +438          raise ENOENT. */
  +439  #ifdef MS_WINDOWS
  +440          if (GetLastError() == ERROR_INVALID_HANDLE) {
  +441              PyErr_SetFromWindowsErr(0);
  +442  #else
  +443          if (errno == EBADF) {
  +444              PyErr_SetFromErrno(PyExc_OSError);
  +445  #endif
  +446              goto error;
  +447          }
  +448      }
  +449      else {
b) afaik, if fd<0 the open() call did not succeed and the old logic will report "something"